<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>One-dimensional Euler equations solver</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QComboBox" name="problem">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>330</width>
      <height>30</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>choose a predifined test case</string>
    </property>
    <property name="currentIndex">
     <number>-1</number>
    </property>
    <item>
     <property name="text">
      <string>TEST 1 (Modified Sod)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TEST 2 (123 Problems)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TEST 3 (Left Woodward &amp; Colella)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TEST 4 (Right Woodward &amp; Colella)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TEST 5 (Collision of 2 shocks)</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Problem:</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="rhol">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>160</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>density at the left side</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="rhor">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>160</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>density at the right side</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="ul">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>190</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>velocity x at the left side</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="ur">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>190</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>velocity x at the right side</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="pl">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>220</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>pressure at the left side</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="pr">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>220</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>pressure at the right side</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="x0">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>120</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>initial position</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="xf">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>120</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>final position</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="pd">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>270</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>position of the diaphragm</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="gamma">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>270</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>heat capacity ratio</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="tout">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>300</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>final time</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="maxt">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>300</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>max number of time iteration</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="cfl">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>330</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>CFL number</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="tol">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>330</y>
      <width>120</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>pressure guess tolerance</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>xi:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>120</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>xf:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>ρL:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>160</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>ρR:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>190</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>uL:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>190</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>uR:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>pL:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>220</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>pR:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>270</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>P.D.:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>270</y>
      <width>20</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>γ:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>300</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>tout:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>300</y>
      <width>41</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Maxt:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>CFL:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>330</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>tol:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="calculate">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>380</y>
      <width>171</width>
      <height>35</height>
     </rect>
    </property>
    <property name="text">
     <string>Calculate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>n:</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="n">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>70</y>
      <width>100</width>
      <height>20</height>
     </rect>
    </property>
    <property name="acceptDrops">
     <bool>true</bool>
    </property>
    <property name="toolTip">
     <string>number of cells</string>
    </property>
    <property name="accessibleDescription">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>60</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>BC1:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>90</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>BC2:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="BC1">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>55</y>
      <width>160</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>boundary condition at the left side</string>
    </property>
    <property name="currentIndex">
     <number>-1</number>
    </property>
    <item>
     <property name="text">
      <string>Transmissive</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Reflective</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="BC2">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>85</y>
      <width>160</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>boundary condition at the right side</string>
    </property>
    <property name="currentIndex">
     <number>-1</number>
    </property>
    <item>
     <property name="text">
      <string>Transmissive</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Reflective</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_19">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>0</y>
      <width>81</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Flux solver:</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>30</y>
      <width>150</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Exact Riemann Problem solver</string>
    </property>
    <property name="text">
     <string>Exact RP solver</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_2">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>60</y>
      <width>150</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Roe solver</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_3">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>90</y>
      <width>150</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>HLL solver</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_4">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>120</y>
      <width>150</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>HLLC solver</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_5">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>150</y>
      <width>150</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Rusanov solver</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QPushButton" name="clean">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>380</y>
      <width>111</width>
      <height>35</height>
     </rect>
    </property>
    <property name="text">
     <string>Clean</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="log">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>180</y>
      <width>261</width>
      <height>221</height>
     </rect>
    </property>
    <property name="toolTip">
     <string/>
    </property>
   </widget>
   <widget class="QCheckBox" name="show">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>355</y>
      <width>341</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>plot the exact solution aside with the numerical solution</string>
    </property>
    <property name="text">
     <string>Show exact solution (only to transmissive BCs)</string>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>640</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup">
   <property name="exclusive">
    <bool>true</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
